engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -442,14 +406,14 @@ public void renderWorld(Camera camera) {
         DefaultRenderingProcess.getInstance().beginRenderLightGeometryStencilPass();
         Material program = Assets.getMaterial("engine:prog.simple");
```

Comment:
Not the asset manager. ShaderManagerLwjgl. These are not automatically generated assets based on a naming convention! These are specific materials generated by the rendering system. They are also purely for the internal usage of the rendering system - these are the materials used by the ShaderParameter classes. Ideally we would eventually phase them out, replaced with a shader variable semantics support or something, but because that is quite a body of work not really worth messing with.

There is not a one to one relationship between materials and shaders, and in general a shader does not have enough information to generate a material (since it doesn't have values for the shader's variables - this is why a material exists). A shader is included in the MaterialData class used to generate a material. So if you want a material for a given shader, you make it.